### PR TITLE
Initial Update to Developer Guide

### PR DESCRIPTION
**Pull request by IsabelChong**
== Ready to Merge ==

- Changes of mentions of `AddressBook` to `CLIpboard` or `Roster` depending on context
- Changes of mentions of `Person` to `Student`
- Updated links to refer to our code instead of `AB3`
- Added in some more use cases
  - Would have to figure out how many use cases we would want to add it, as it would be a lot to add in all
- Did not touch the section under `Implementation` as major update would be required

Initial update to main issue #220

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -296,82 +296,131 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* 3a. The student does not exist in the list
+    Use case ends.
 
-    * 3a1. CLipboard shows an error message.
+**Extensions**
 
-      Use case resumes at step 2.
+* Similar to UC2 extension, for student index
+
+**Use case 7: Update a remark under a student's profile**
```
Comment (tyx021) [inline docs/DeveloperGuide.md:410]: Inconsistent header font size, maybe change it to `<h3>Use case 7: Update a remark under a student's profile</h3>`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -296,82 +296,131 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Preconditions: User is on the Student Page**
+
+**MSS**
+
+3.  User requests to update a remark under a specific student in the list
+4.  CLIpboard updates the particulars of the student to the new remark
```
Comment (tyx021) [inline docs/DeveloperGuide.md:417]: Is it intended to have numbering or 3 and 4 here?

**Review by tyx021 — COMMENTED** [review]
Good job! 👍😄

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (IsabelChong) [reply docs/DeveloperGuide.md:410]: Thanks for catching that!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (IsabelChong) [reply docs/DeveloperGuide.md:417]: Thanks for catching this! Updated the numbering to 1 and 2.

**Pull request merged**